### PR TITLE
fix: forEach on all of the actions

### PR DESCRIPTION
**Pull request by NetaMolcho**

### CODE REVIEW

File: src/actions/controller.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ export class ActionsController {
     public static loadActions() {
       vscode.extensions.all.forEach((extension) => {
         if (extension?.packageJSON?.BASContributes?.actions) {
-          this.actions.push(extension.packageJSON.BASContributes.actions);
+          const newActions: IAction[] = extension?.packageJSON?.BASContributes?.actions;
+          newActions.forEach((action) => {
+            this.actions.push(action);
+          });
```
Comment (tomer-epstein) [inline src/actions/controller.ts:14]: extension.packageJSON.BASContributes.actions.forEach((action) => {
            this.actions.push(action);
          });

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

File: src/actions/controller.ts (hunk 1 above)
Comment (NetaMolcho) [reply src/actions/controller.ts:14]: it needs to know that this is IAction[] type, it doesn't accept it without a declaration

**Review by tomer-epstein — CHANGES_REQUESTED** [review]
increment app-studio-toolkit version to 0.0.6

**Review by tomer-epstein — DISMISSED** [review]

File: src/actions/controller.ts (hunk 1 above)
Comment (tomer-epstein) [reply src/actions/controller.ts]: > it needs to know that this is IAction[] type, it doesn't accept it without a declaration

(extension.packageJSON.BASContributes.actions as IAction[]).forEach

File: src/actions/controller.ts [hunk 2]
```diff
@@ -8,7 +8,9 @@ export class ActionsController {
     public static loadActions() {
       vscode.extensions.all.forEach((extension) => {
         if (extension?.packageJSON?.BASContributes?.actions) {
-          this.actions.push(extension.packageJSON.BASContributes.actions);
+          (extension.packageJSON.BASContributes.actions as IAction[]).forEach((action) => {
+            this.actions.push(action);
+          });
         }
```
Comment (tomer-epstein) [inline src/actions/controller.ts:14]: Can you please add a test for this method?

File: tests/api.spec.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ import * as _ from "lodash";
 import * as sinon from "sinon";
 import { IAction, ActionType } from "../src/actions/interfaces";
 import { ActionsController } from '../src/actions/controller';
+import * as vscode from "vscode";
```
Comment (tomer-epstein) [inline tests/api.spec.ts:8]: Please remove.
We are using --> mockVscode

File: tests/api.spec.ts [hunk 4]
```diff
@@ -77,6 +79,30 @@ describe("api unit test", () => {
...
+                    }]
+                },
+            }
+        }]
+        _.set(vscode, "extensions.all", allExtensioms);
+
+        const action: IAction = {
+			"id" : "abc123",
+			"actionType" : ActionType.Command,
+		}
```
Comment (tomer-epstein) [inline tests/api.spec.ts:98]: const action: IAction = {
			"id" : "abc123",
			"actionType" : ActionType.Command,
		}
        const allExtensioms = [{
            packageJSON: {
                "BASContributes": {
                    "actions": [action]
                },
            }
        }]
        _.set(vscode, "extensions.all", allExtensioms);

**Review by tomer-epstein — CHANGES_REQUESTED** [review]

File: tests/api.spec.ts (hunk 4 above)
Comment (NetaMolcho) [reply tests/api.spec.ts:98]: you have the vscode import for this set

File: tests/api.spec.ts (hunk 3 above)
Comment (NetaMolcho) [reply tests/api.spec.ts:8]: it's impossible to do set on mockVscode

File: tests/api.spec.ts (hunk 4 above)
Comment (NetaMolcho) [reply tests/api.spec.ts:98]: allExtensions should resemble package.json, so it should contain strings

Comment (NetaMolcho) [reply tests/api.spec.ts:98]: allExtension should resemble package.json, so it should only contain strings

**Review by tomer-epstein — APPROVED** [review]

**Pull request merged**